src/main/java/no/ntnu/mycbr/rest/service/EphemeralService.java

```diff
@@ -0,0 +1,247 @@
+package no.ntnu.mycbr.rest.service;
+
+import static no.ntnu.mycbr.rest.utils.QueryUtils.getFullResult;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import no.ntnu.mycbr.core.DefaultCaseBase;
+import no.ntnu.mycbr.core.ICaseBase;
+import no.ntnu.mycbr.core.Project;
+import no.ntnu.mycbr.core.casebase.Attribute;
+import no.ntnu.mycbr.core.casebase.Instance;
+import no.ntnu.mycbr.core.model.AttributeDesc;
+import no.ntnu.mycbr.core.model.Concept;
+import no.ntnu.mycbr.core.model.SymbolDesc;
+import no.ntnu.mycbr.core.retrieval.Retrieval;
+import no.ntnu.mycbr.core.retrieval.Retrieval.RetrievalCustomer;
+import no.ntnu.mycbr.core.similarity.Similarity;
+import no.ntnu.mycbr.util.Pair;
+import no.ntnu.mycbr.rest.App;
+import no.ntnu.mycbr.rest.Case;
+import no.ntnu.mycbr.rest.Query;
+import no.ntnu.mycbr.rest.utils.TemporaryAmalgamFctManager;
+import no.ntnu.mycbr.rest.utils.TemporaryAmalgamFctNotChangedException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * The EphemeralService facilitates retrieval an ephemeral case base including Self-Similarity retrieval.
+ * @author Amar Jaiswal
+ * @since Nov 24, 2019
+ */
+//s@Service
+public class EphemeralService implements RetrievalCustomer{
+
+    private final Log logger = LogFactory.getLog(getClass());
+    
+    // The name used for ephemeral case base
+    private static final String EPHEMERAL_CASEBASE_NAME = "ephemeral_casebase";
```

Comment:
This looks thread-unsafe. If there are several request at the same time they will try to access/create an ephemeral case base with the same name.
These requests potentially take more time. Thus they have higher chance of access/modification conflict.